### PR TITLE
MODINVSTOR-818: Add new note type

### PR DESCRIPTION
**Pull request by Shans-Kaluhin**
### Purpose

This story is needed to complete the mapping rules for the Authority object from the Marc records (667 field).
In this story we have to add a new type of note "NONPUBLIC GENERAL NOTE"
https://wiki.folio.org/display/FOLIJET/Search+MARC+Authorities

### Approach
 The note types are stored in a json files and get populated in the database when the tenant setup happens. 

  1) Create Json file for the new note similarly to others
  2) Create provided endpoints:
      - GET, POST '/authority-note-types'
      - GET, DELETE, PUT '/authority-note-types/{id}' 

  3) Create new table 'authority_note_type'
  4) Add unit tests

### Learn
[MODINVSTOR-818](https://issues.folio.org/browse/MODINVSTOR-818)
[Holdings note type A{I](https://s3.amazonaws.com/foliodocs/api/mod-inventory-storage/holdings-note-type.html)

### CODE REVIEW

File: reference-data/authority-note-types/NonpublicGeneralNote.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
```
[the remark this answers is not in the document]
Comment (psmagin) [reply reference-data/authority-note-types/NonpublicGeneralNote.json:1]: Do we have another ticket to add other note types that will be needed for marc authority mapping?

[the remark this answers is not in the document]
Comment (Shans-Kaluhin) [reply reference-data/authority-note-types/NonpublicGeneralNote.json:1]: Nope

**Review by Igor-Gorchakov — APPROVED** [review]

File: src/main/java/org/folio/rest/impl/AuthorityNoteTypeAPI.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  @Validate
+  public void getAuthorityNoteTypesById(String id, String lang, Map<String, String> okapiHeaders,
+      Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
+    PgUtil.getById(REFERENCE_TABLE, AuthorityNoteType.class, id,
+      okapiHeaders, vertxContext, GetAuthorityNoteTypesByIdResponse.class, asyncResultHandler);
+  }
+
+  @Override
+  @Validate
+  public void deleteAuthorityNoteTypesById(String id, String lang, Map<String, String> okapiHeaders,
```
Comment (oleksandr-dekin) [inline src/main/java/org/folio/rest/impl/AuthorityNoteTypeAPI.java:44]: Do we need to restrict deletion default note types?

**Review by natali-zaitseva — APPROVED** [review]

File: src/main/java/org/folio/rest/impl/AuthorityNoteTypeAPI.java (hunk 2 above)
Comment (Shans-Kaluhin) [reply src/main/java/org/folio/rest/impl/AuthorityNoteTypeAPI.java:44]: Khalila said is not necessarily

**Pull request merged**